src/test/e2e/api/proxy/proxy.e2e.test.ts

```diff
@@ -982,6 +1006,133 @@ test('should return all features when specified', async () => {
+        .get('/api/frontend')
+        .set('Authorization', frontendToken.secret)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((res) => {
+            expect(res.body).toEqual({
+                toggles: [
+                    {
+                        name: 'enabledFeature',
+                        enabled: true,
+                        impressionData: false,
+                        variant: { enabled: false, name: 'disabled' },
+                    },
+                    {
+                        name: 'disabledFeature',
+                        enabled: false,
+                        impressionData: false,
+                        variant: { enabled: false, name: 'disabled' },
+                    },
+                ],
+            });
+        });
+});
+test('should not return all features if both feature flags are not true', async () => {
+    if (app.config.experimental != null) {
+        app.config.experimental.flags.proxyReturnAllTogglesKillSwitch = true;
+        app.config.experimental.flags.proxyReturnAllToggles = false;
+    }
+    const frontendToken = await createApiToken(ApiTokenType.FRONTEND);
+    await createFeatureToggle({
+        name: 'enabledFeature1',
+        enabled: true,
+        strategies: [{ name: 'default', constraints: [], parameters: {} }],
+    });
+    await createFeatureToggle({
+        name: 'enabledFeature2',
+        enabled: true,
+        strategies: [
+            {
+                name: 'flexibleRollout',
+                constraints: [],
+                parameters: {
+                    rollout: '100',
+                    stickiness: 'default',
+                    groupId: 'some-new',
+                },
+            },
+        ],
+    });
+    await createFeatureToggle({
+        name: 'disabledFeature',
+        enabled: true,
+        strategies: [
+            {
+                name: 'flexibleRollout',
+                constraints: [],
+                parameters: {
+                    rollout: '0',
+                    stickiness: 'default',
+                    groupId: 'some-new',
+                },
+            },
+        ],
+    });
+    await app.request
+        .get('/api/frontend')
+        .set('Authorization', frontendToken.secret)
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect((res) => {
+            expect(res.body).toEqual({
+                toggles: [
+                    {
+                        name: 'enabledFeature1',
+                        enabled: true,
+                        impressionData: false,
+                        variant: { enabled: false, name: 'disabled' },
+                    },
+                    {
+                        name: 'enabledFeature2',
+                        enabled: true,
+                        impressionData: false,
+                        variant: { enabled: false, name: 'disabled' },
+                    },
+                ],
+            });
```

Comment:
Just a thought I had (and not something that needs to be addressed in this PR or maybe ever):

Do these tests test too much? The test name only indicates that it returns the corrects toggles, and not necessarily that they contain the right kind of data (other than status). That makes them kind of brittle in case we change the payload later (by adding more data, like how we did for `impressionData`). If this test is only to test the name, would it be better to do something like this?

```suggestion
            expect(res.body.toggles.map(toggle => toggle.name).toEqual(["enabledFeature1", "enabledFeature2"]);
```

(The same thing could probably be done to separate the duplicate testing that we discussed earlier 💁🏼 )